src/chapters/aspdotnet-model-classes/studio.rst

```diff
@@ -0,0 +1,110 @@
+
+The starter code has been modified from where you left off last class. Grab the refactored code 
+on the `user-signup-starter branch <https://github.com/LaunchCodeEducation/SpaDay/tree/user-signup-starter>`__. 
+
+You'll notice in this branch that the name field has been removed from the service selection form. Once we
+implement user-signup functionality, we can use a given user's name to identify the spa client. We've also 
+moved data into a ``Client`` model and out of the ``SpaController`` class.
+
+In this studio, we'll ask you to write another model, ``User``. ``User`` and ``Client`` may at first 
+appear redundant, but in the future as you develop your spa application, you may find a scenario where 
+a user is logging in who is not also ``Client``.
+
+Getting Ready
+-------------
+
+Within ``SpaDay``, create the following files. 
+
+#. Create a ``UserController`` in ``Controllers``.
+#. Create a new folder, ``User``, within ``Views`` 
+#. Create ``Index.cshtml`` and ``Add.cshtml`` templates within ``Views/User/`` 
+#. Create a ``User`` class within ``Models``
+
+Creating the Model
+------------------
+
+Your ``User`` class should have a few properties: ``Username``, ``Email``, and ``Password``. 
+
+Rendering the Add User Form
+---------------------------
+
+#. In the ``UserController``, create an action method ``Add()`` to
+   render the form. This handler should correspond to the path
+   ``/user/add``, and for now, it can just return the ``Add.cshtml`` view.
+
+   .. admonition:: Tip
+
+      Don't forget to add ``/user/add`` to your path when you test your new features. 
+
+#. Within the ``Add.cshtml`` template, create a form that accepts inputs for
+   each of the ``User`` class properties. Include an additional password input field to verify 
+   the password input. The form should be set up to ``POST`` to ``/user``. 
+
+#. Be sure to set ``type="password"`` for the password and verify inputs,
+   to ensure the passwords are not visible when being typed into the form.
+   You can also set ``type="email"`` on the email input, which will enable
+   some basic client-side validation. We'll tackle validation in more detail 
+   in the next studio. 
+
+Processing Form Submission
+--------------------------
+
+#. Within the ``UserController``, create an action method with this signature:
+
+   .. sourcecode:: csharp
+
+      public IActionResult AddUser(User newUser, string verify) {
+         // add form submission handling code here
+      }
+
+   This will use model binding to create a new user object, ``newUser``, and
+   pass it into your action method. 
+
+   .. admonition :: Note
+   
+      You don’t need to store the ``User`` object anywhere for this studio.
+      We’re focusing on form handling and validation in this exercise. If you
+      want to keep track of users using the method we employed in the models
+      lesson video, check out the Bonus Missions below.
+
+#. Check that the ``verify`` parameter matches the
```

Comment:
The studio looks good until some roadblocks around this step - 

the instructions above tell us to have the add user form post to /user. Which I think ultimately it should design-wise. we want that user index view to be the "welcome to the spa" message. the updated solution now though has the form posting to /user/add. 
I'm also thinking you can completely avoid that redirect viewbag issue by not returning a redirect and instead returning the View("Index"). Since we do briefly touch upon returning a different view than the default using an argument, this is a good opportunity to take advantage of it, no? It works and requires less workaround-code and less code overall than returning a redirect. A solution that returns View("Index") is more straightforward imo but i'd like to hear your opinion. And if its not clear, if you skip the redirect here, then you can just assign the viewbag username or name property in this method.

Reply:
the other thing is that I'd recommend renaming this action method based off of what we talked about in last week's working meeting. something more descriptive like, "processAddUserForm" or w/e. and i'd also say highly recommend we give them the instructions that this method will need information to respond to the add form post action and will need an explicit route attribute as well. maybe we don't need to include exactly what these look like in the sourcecode directive but some direction is a good idea - otherwise as I walk through the instructions myself, im worried these bits may get lost

Reply:
we definitely don't cover that to your second point. What I mean is to return View("Index")